Pomchecker CLI does not resolve artifacts from Maven Central

Having a pom.xml like the following: 

```xml
<?xml version="1.0" encoding="UTF-8"?>
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
    <modelVersion>4.0.0</modelVersion>

    <parent>
        <groupId>io.quarkiverse</groupId>
        <artifactId>quarkiverse-parent</artifactId>
        <version>17</version>
    </parent>
    <groupId>io.quarkiverse.slack</groupId>
    <artifactId>quarkus-slack-parent</artifactId>
    <version>0.0.2</version>
    <packaging>pom</packaging>
</project>
```

First run `rm -rf ~/.m2/repository/io/quarkiverse` to make sure your local repository does not have the parent artifact. 

When I run `pomchecker check-maven-central --file pom.xml` the following error is shown:

```bash
[INFO] Maven Central checks: /private/tmp/jreleaser/repository/pom.xml
org.apache.maven.project.ProjectBuildingException: 1 problem was encountered while building the effective model for io.quarkiverse.slack:quarkus-slack-parent:0.0.2
[FATAL] Non-resolvable parent POM for io.quarkiverse.slack:quarkus-slack-parent:0.0.2: The following artifacts could not be resolved: io.quarkiverse:quarkiverse-parent:pom:17 (absent): Could not find artifact io.quarkiverse:quarkiverse-parent:pom:17 and 'parent.relativePath' points at wrong local POM @ line 5, column 13
 for project io.quarkiverse.slack:quarkus-slack-parent:0.0.2 at /private/tmp/jreleaser/repository/pom.xml
java.lang.IllegalStateException: org.apache.maven.project.ProjectBuildingException: 1 problem was encountered while building the effective model for io.quarkiverse.slack:quarkus-slack-parent:0.0.2
[FATAL] Non-resolvable parent POM for io.quarkiverse.slack:quarkus-slack-parent:0.0.2: The following artifacts could not be resolved: io.quarkiverse:quarkiverse-parent:pom:17 (absent): Could not find artifact io.quarkiverse:quarkiverse-parent:pom:17 and 'parent.relativePath' points at wrong local POM @ line 5, column 13
 for project io.quarkiverse.slack:quarkus-slack-parent:0.0.2 at /private/tmp/jreleaser/repository/pom.xml
	at org.kordamp.maven.checker.cli.internal.PomParser.createMavenProject(PomParser.java:102)
	at org.kordamp.maven.checker.cli.internal.PomParser.createMavenProject(PomParser.java:66)
	at org.kordamp.maven.checker.cli.CheckMavenCentral.execute(CheckMavenCentral.java:61)
	at org.kordamp.maven.checker.cli.AbstractCommand.call(AbstractCommand.java:102)
	at org.kordamp.maven.checker.cli.CheckMavenCentral.call(CheckMavenCentral.java:30)
	at org.kordamp.maven.checker.cli.AbstractCommand.call(AbstractCommand.java:34)
	at picocli.CommandLine.executeUserObject(CommandLine.java:2041)
	at picocli.CommandLine.access$1500(CommandLine.java:148)
	at picocli.CommandLine$RunLast.executeUserObjectOfLastSubcommandWithSameParent(CommandLine.java:2461)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2453)
	at picocli.CommandLine$RunLast.handle(CommandLine.java:2415)
	at picocli.CommandLine$AbstractParseResultHandler.execute(CommandLine.java:2273)
	at picocli.CommandLine$RunLast.execute(CommandLine.java:2417)
	at picocli.CommandLine.execute(CommandLine.java:2170)
	at org.kordamp.maven.checker.cli.Main.execute(Main.java:86)
	at org.kordamp.maven.checker.cli.Main.run(Main.java:72)
	at org.kordamp.maven.checker.cli.Main.main(Main.java:64)
Caused by: org.apache.maven.project.ProjectBuildingException: 1 problem was encountered while building the effective model for io.quarkiverse.slack:quarkus-slack-parent:0.0.2
[FATAL] Non-resolvable parent POM for io.quarkiverse.slack:quarkus-slack-parent:0.0.2: The following artifacts could not be resolved: io.quarkiverse:quarkiverse-parent:pom:17 (absent): Could not find artifact io.quarkiverse:quarkiverse-parent:pom:17 and 'parent.relativePath' points at wrong local POM @ line 5, column 13
 for project io.quarkiverse.slack:quarkus-slack-parent:0.0.2 at /private/tmp/jreleaser/repository/pom.xml
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:180)
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:123)
	at org.kordamp.maven.checker.cli.internal.PomParser.createMavenProject(PomParser.java:99)
	... 16 more
Caused by: org.apache.maven.model.building.ModelBuildingException: 1 problem was encountered while building the effective model for io.quarkiverse.slack:quarkus-slack-parent:0.0.2
[FATAL] Non-resolvable parent POM for io.quarkiverse.slack:quarkus-slack-parent:0.0.2: The following artifacts could not be resolved: io.quarkiverse:quarkiverse-parent:pom:17 (absent): Could not find artifact io.quarkiverse:quarkiverse-parent:pom:17 and 'parent.relativePath' points at wrong local POM @ line 5, column 13

	at org.apache.maven.model.building.DefaultModelProblemCollector.newModelBuildingException(DefaultModelProblemCollector.java:176)
	at org.apache.maven.model.building.DefaultModelBuilder.readParentExternally(DefaultModelBuilder.java:1033)
	at org.apache.maven.model.building.DefaultModelBuilder.readParent(DefaultModelBuilder.java:801)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:327)
	at org.apache.maven.model.building.DefaultModelBuilder.build(DefaultModelBuilder.java:243)
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:176)
	... 18 more
```

--- a/pomchecker-cli/pom.xml
+++ b/pomchecker-cli/pom.xml
@@ -145,6 +145,12 @@
             <artifactId>slf4j-simple</artifactId>
             <scope>runtime</scope>
         </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter</artifactId>
+            <version>5.11.0</version>
+            <scope>test</scope>
+        </dependency>
     </dependencies>
 
     <build>

--- a/pomchecker-cli/src/main/java/org/kordamp/maven/checker/cli/internal/PomParser.java
+++ b/pomchecker-cli/src/main/java/org/kordamp/maven/checker/cli/internal/PomParser.java
@@ -22,6 +22,8 @@ import com.google.common.collect.ImmutableMap;
 import eu.maveniverse.maven.mima.context.Context;
 import eu.maveniverse.maven.mima.context.ContextOverrides;
 import eu.maveniverse.maven.mima.context.Runtimes;
+import org.apache.maven.artifact.repository.MavenArtifactRepository;
+import org.apache.maven.artifact.repository.layout.DefaultRepositoryLayout;
 import org.apache.maven.execution.DefaultMavenExecutionRequest;
 import org.apache.maven.execution.MavenExecutionRequest;
 import org.apache.maven.project.MavenProject;
@@ -37,12 +39,12 @@ import org.codehaus.plexus.PlexusContainer;
 import org.codehaus.plexus.PlexusContainerException;
 import org.codehaus.plexus.classworlds.ClassWorld;
 import org.codehaus.plexus.component.repository.exception.ComponentLookupException;
+import org.eclipse.aether.repository.RemoteRepository;
 
 import java.io.File;
 import java.util.Locale;
 import java.util.Properties;
-
-import static java.util.Objects.requireNonNull;
+import java.util.stream.Collectors;
 
 /**
  * @author Andres Almiray
@@ -83,7 +85,8 @@ public class PomParser {
                 mavenExecutionRequest.getProjectBuildingRequest();
 
             projectBuildingRequest.setRepositorySession(context.repositorySystemSession());
-
+            projectBuildingRequest.setRemoteRepositories(context.remoteRepositories()
+                    .stream().map(PomParser::toArtifactRepository).collect(Collectors.toList()));
             // Profile activation needs properties such as JDK version
             Properties properties = new Properties(); // allowing duplicate entries
             properties.putAll(projectBuildingRequest.getSystemProperties());
@@ -137,5 +140,13 @@ public class PomParser {
             default:
                 return "x86_32";
         }
+    }
+
+    private static MavenArtifactRepository toArtifactRepository(RemoteRepository remoteRepository) {
+        MavenArtifactRepository mavenArtifactRepository = new MavenArtifactRepository();
+        mavenArtifactRepository.setId(remoteRepository.getId());
+        mavenArtifactRepository.setUrl(remoteRepository.getUrl());
+        mavenArtifactRepository.setLayout(new DefaultRepositoryLayout());
+        return mavenArtifactRepository;
     }
 }

--- a/pomchecker-cli/src/test/java/org/kordamp/maven/checker/cli/internal/PomParserTest.java
+++ b/pomchecker-cli/src/test/java/org/kordamp/maven/checker/cli/internal/PomParserTest.java
@@ -1,0 +1,36 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Copyright 2020-2024 Andres Almiray.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.kordamp.maven.checker.cli.internal;
+
+import org.apache.maven.project.MavenProject;
+import org.junit.jupiter.api.Test;
+
+import java.io.File;
+import java.net.URL;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+class PomParserTest {
+
+    @Test
+    void parse() throws Exception {
+        URL resource = getClass().getClassLoader().getResource("test-pom.xml");
+        MavenProject mavenProject = PomParser.createMavenProject(new File(resource.toURI()));
+        assertEquals("quarkus-slack-parent",mavenProject.getArtifactId());
+    }
+}

--- a/pomchecker-cli/src/test/resources/test-pom.xml
+++ b/pomchecker-cli/src/test/resources/test-pom.xml
@@ -1,0 +1,34 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    SPDX-License-Identifier: Apache-2.0
+
+    Copyright 2020-2024 Andres Almiray.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        https://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 https://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <groupId>io.quarkiverse</groupId>
+        <artifactId>quarkiverse-parent</artifactId>
+        <version>17</version>
+    </parent>
+    <groupId>io.quarkiverse.slack</groupId>
+    <artifactId>quarkus-slack-parent</artifactId>
+    <version>0.0.2</version>
+    <packaging>pom</packaging>
+</project>
+

